static/js/components/email/hoc.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import R from 'ramda';
+
+import EmailCompositionDialog from './EmailCompositionDialog';
+import { getAllEmailProperties } from './lib';
+import { getDisplayName } from '../../util/util';
+
+export const withEmailDialog = R.curry((emailType, WrappedComponent) => {
+  class WithEmailDialog extends React.Component {
+    constructor(props) {
+      super(props);
+      this.renderCompositionDialog = this.renderCompositionDialog.bind(this);
+      this.allEmailProperties = getAllEmailProperties();
+      this.emailType = emailType;
+    }
+
+    renderCompositionDialog() {
+      let {
+        dispatch,
+        ui: { dialogVisibility },
+        email
+      } = this.props;
+      return <EmailCompositionDialog
+        dispatch={dispatch}
+        dialogVisibility={dialogVisibility}
+        email={email}
+        emailProperties={this.allEmailProperties}
+      />;
+    }
+
+    render() {
+      let emailDialog = this.props.emailDialog || this.renderCompositionDialog();
+
+      let emailOpenActions = this.props.emailOpenActions || {};
+      emailOpenActions[this.emailType] = R.partial(
+        this.allEmailProperties[this.emailType].emailOpenAction, [this.props.dispatch]
+      );
+
+      return <WrappedComponent
+        {...this.props}
+        emailDialog={emailDialog}
+        emailOpenActions={emailOpenActions}
+      />;
+    }
+  }
+
+  if (WithEmailDialog.displayName && !WithEmailDialog.displayName.startsWith('WithEmailDialogs')) {
```

Comment:
Not reviewing the PR but just spotted this. Isn't `displayName` always `WithEmailDialogs` at this point?